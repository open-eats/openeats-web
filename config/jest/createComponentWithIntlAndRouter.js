import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { MemoryRouter } from 'react-router-dom'

const createComponentWithIntl = (children, props = { locale: 'en' }) => {
  return renderer.create(
    <IntlProvider {...props}>
      <MemoryRouter>
        { children }
      </MemoryRouter>
    </IntlProvider>
  );
};

export default createComponentWithIntl;