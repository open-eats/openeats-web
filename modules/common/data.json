{
  "list": {
    "lists": [
      {
        "id": 2,
        "title": "fgh",
        "slug": "fgh",
        "pub_date": "",
        "author": "ryan",
        "item_count": 2,
        "items": [
          {
            "id": 10,
            "list": 2,
            "title": "ert",
            "completed": false
          },
          {
            "id": 11,
            "list": 2,
            "title": "dfg",
            "completed": false
          }
        ]
      },
      {
        "id": 3,
        "title": "dfg",
        "slug": "dfg",
        "pub_date": "",
        "author": "ryan",
        "item_count": 2,
        "items": [
          {
            "id": 12,
            "list": 3,
            "title": "ert",
            "completed": false
          },
          {
            "id": 13,
            "list": 3,
            "title": "ertertertertertert",
            "completed": false
          }
        ]
      },
      {
        "id": 4,
        "title": "fghfgh",
        "slug": "fghfgh",
        "pub_date": "",
        "author": "ryan",
        "item_count": 0,
        "items": []
      },
      {
        "id": 5,
        "title": "345",
        "slug": "345",
        "pub_date": "",
        "author": "ryan",
        "item_count": 0,
        "items": []
      },
      {
        "id": 6,
        "title": "6",
        "slug": "6",
        "pub_date": "",
        "author": "ryan",
        "item_count": 4,
        "items": [
          {
            "id": 14,
            "list": 6,
            "title": "fgh",
            "completed": false
          },
          {
            "id": 15,
            "list": 6,
            "title": "try",
            "completed": false
          },
          {
            "id": 16,
            "list": 6,
            "title": "rt",
            "completed": false
          },
          {
            "id": 17,
            "list": 6,
            "title": "e5ry",
            "completed": false
          }
        ]
      }
    ],
    "error": ""
  },
  "user": {
    "id": 1,
    "token": "fghjfghj"
  },
  "news": [
    {
      "id": 1,
      "title": "OpenEats 2",
      "content": "OpenEats2 is an open source recipe management site. You can share recipes with friends, rate recipes, store your favorite recipes to find easily, and more. If you want to run your own personal OpenEats site visit the <a href=\"https://github.com/ryannoelk/OpenEats\" target=\"_blank\">Github page</a>",
      "frontpage": true,
      "image": null
    }
  ],
  "browse": {
    "search": {
      "results": {
        "limit=12": {
          "recipes": [
            {
              "id": 34,
              "photo_thumbnail": "http://localhost:8000/site-media/CACHE/images/upload/recipe_photos/Sheet-Pan-BBQ-Meatloaf-Dinner-V2_Iyritb3/c3eaf4db4b2bc29e514a92607ddb4904.jpg",
              "pub_date": "2018-01-30",
              "title": "dgf",
              "info": "dfg",
              "rating": 5
            },
            {
              "id": 33,
              "photo_thumbnail": "http://localhost:8000/site-media/CACHE/images/upload/recipe_photos/g3O1DsG/c08dba895cc303fa83d51c6d95463a28.jpg",
              "pub_date": "2018-01-30",
              "title": "zdfgsdfg",
              "info": "234234",
              "rating": 4
            },
            {
              "id": 32,
              "photo_thumbnail": "http://localhost:8000/site-media/CACHE/images/upload/recipe_photos/1494520435526/bb0a685360f8435d57933938958cdc4b.jpeg",
              "pub_date": "2018-01-30",
              "title": "my recipe",
              "info": "asdfgsdgsdg",
              "rating": 4
            },
            {
              "id": 31,
              "photo_thumbnail": "http://localhost:8000/site-media/CACHE/images/upload/recipe_photos/Sheet-Pan-BBQ-Meatloaf-Dinner-V2/e50552cc0c3c43ef2b79d02f2a16e892.jpg",
              "pub_date": "2018-01-29",
              "title": "new recipe",
              "info": "asdfasdf",
              "rating": 3
            },
            {
              "id": 25,
              "photo_thumbnail": "http://localhost:8000/site-media/CACHE/images/upload/recipe_photos/Wild-Rice-and-Vegetable-Casserole-open/3d154ad662593b62093ef132c956e24c.jpg",
              "pub_date": "2011-05-21",
              "title": "Tasty Ch124ili 4",
              "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
              "rating": 3
            },
            {
              "id": 26,
              "photo_thumbnail": "http://localhost:8000/site-media/CACHE/images/upload/recipe_photos/20161119_190343/8150a7802d8c7d4a9901aae42820d808.jpg",
              "pub_date": "2011-05-21",
              "title": "Tasty Ch14ili 4",
              "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
              "rating": 3
            },
            {
              "id": 27,
              "photo_thumbnail": null,
              "pub_date": "2011-05-21",
              "title": "Tasty Ch15ili 4",
              "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
              "rating": 3
            },
            {
              "id": 30,
              "photo_thumbnail": "http://localhost:8000/site-media/CACHE/images/upload/recipe_photos/spicy-honey-vinaigrette-1A29a-1-P3060029-300x300/ca2abcb7cd4eafaa43448d293b47b7be.jpg",
              "pub_date": "2011-05-21",
              "title": "Tasty Ch345ili 4",
              "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
              "rating": 3
            },
            {
              "id": 23,
              "photo_thumbnail": null,
              "pub_date": "2011-05-21",
              "title": "Tasty Chi124li 4",
              "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
              "rating": 3
            },
            {
              "id": 21,
              "photo_thumbnail": null,
              "pub_date": "2011-05-21",
              "title": "Tasty Chil12i 4",
              "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
              "rating": 3
            },
            {
              "id": 1,
              "photo_thumbnail": null,
              "pub_date": "2011-05-21",
              "title": "Tasty Chili",
              "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
              "rating": 3
            },
            {
              "id": 7,
              "photo_thumbnail": null,
              "pub_date": "2011-05-21",
              "title": "Tasty Chili 1114",
              "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
              "rating": 3
            }
          ],
          "totalRecipes": 30
        }
      },
      "loading": false
    },
    "filters": {
      "courses": {
        "results": {
          "limit=12": [
            {
              "id": 3,
              "total": 1,
              "title": "Appetizer",
              "slug": "appetizer",
              "author": 1
            },
            {
              "id": 1,
              "total": 1,
              "title": "Breakfast",
              "slug": "breakfast",
              "author": 1
            },
            {
              "id": 4,
              "total": 2,
              "title": "Dessert",
              "slug": "dessert",
              "author": 1
            },
            {
              "id": 2,
              "total": 26,
              "title": "Entry",
              "slug": "entry",
              "author": 1
            }
          ]
        },
        "loading": false,
        "error": false
      },
      "cuisines": {
        "results": {
          "limit=12": [
            {
              "id": 1,
              "total": 26,
              "title": "American",
              "slug": "american",
              "author": 1
            },
            {
              "id": 3,
              "total": 1,
              "title": "French",
              "slug": "french",
              "author": 1
            },
            {
              "id": 5,
              "total": 1,
              "title": "Italian",
              "slug": "italian",
              "author": 1
            },
            {
              "id": 9,
              "total": 2,
              "title": "Thai",
              "slug": "thai",
              "author": 1
            }
          ]
        },
        "loading": false,
        "error": false
      },
      "ratings": {
        "results": {
          "limit=12": [
            {
              "rating": 5,
              "total": 1,
              "slug": 5,
              "title": "5 stars"
            },
            {
              "rating": 4,
              "total": 2,
              "slug": 4,
              "title": "4 stars"
            },
            {
              "rating": 3,
              "total": 27,
              "slug": 3,
              "title": "3 stars"
            }
          ]
        },
        "loading": false,
        "error": false
      }
    },
    "miniBrowse": [
      {
        "id": 33,
        "title": "zdfgsdfg",
        "pub_date": "2018-01-30",
        "rating": 4,
        "photo_thumbnail": "http://localhost:8000/site-media/CACHE/images/upload/recipe_photos/g3O1DsG/c08dba895cc303fa83d51c6d95463a28.jpg",
        "info": "234234"
      },
      {
        "id": 25,
        "title": "Tasty Ch124ili 4",
        "pub_date": "2011-05-21",
        "rating": 3,
        "photo_thumbnail": "http://localhost:8000/site-media/CACHE/images/upload/recipe_photos/Wild-Rice-and-Vegetable-Casserole-open/3d154ad662593b62093ef132c956e24c.jpg",
        "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma"
      },
      {
        "id": 3,
        "title": "Tasty Chili 3",
        "pub_date": "2011-05-21",
        "rating": 3,
        "photo_thumbnail": null,
        "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma"
      },
      {
        "id": 16,
        "title": "Tasty Chili 444",
        "pub_date": "2011-05-21",
        "rating": 3,
        "photo_thumbnail": null,
        "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma"
      }
    ]
  },
  "recipe": {
    "recipes": [
      {
        "id": 25,
        "photo": "http://localhost:8000/site-media/upload/recipe_photos/Wild-Rice-and-Vegetable-Casserole-open.jpg",
        "photo_thumbnail": "http://localhost:8000/site-media/CACHE/images/upload/recipe_photos/Wild-Rice-and-Vegetable-Casserole-open/3d154ad662593b62093ef132c956e24c.jpg",
        "ingredient_groups": [
          {
            "id": 7,
            "title": "",
            "ingredients": [
              {
                "id": 21,
                "quantity": 0,
                "measurement": null,
                "title": "fg"
              }
            ]
          }
        ],
        "tags": [],
        "subrecipes": [],
        "pub_date": "2011-05-21",
        "update_date": "2018-01-30",
        "username": "ryan",
        "title": "Tasty Ch124ili 4",
        "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
        "directions": "fghfgh",
        "source": "",
        "prep_time": 60,
        "cook_time": 60,
        "servings": 8,
        "rating": 3,
        "public": true,
        "author": 1,
        "cuisine": 1,
        "course": 2
      },
      {
        "id": 21,
        "photo": null,
        "photo_thumbnail": null,
        "ingredient_groups": [],
        "tags": [],
        "subrecipes": [],
        "pub_date": "2011-05-21",
        "update_date": "2011-05-21",
        "username": "ryan",
        "title": "Tasty Chil12i 4",
        "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
        "directions": "",
        "source": "",
        "prep_time": 60,
        "cook_time": 60,
        "servings": 8,
        "rating": 3,
        "public": true,
        "author": 1,
        "cuisine": 1,
        "course": 2
      },
      {
        "id": 5,
        "photo": null,
        "photo_thumbnail": null,
        "ingredient_groups": [],
        "tags": [],
        "subrecipes": [],
        "pub_date": "2011-05-21",
        "update_date": "2011-05-21",
        "username": "ryan",
        "title": "Tasty Chili 14",
        "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
        "directions": "",
        "source": "",
        "prep_time": 60,
        "cook_time": 60,
        "servings": 8,
        "rating": 3,
        "public": true,
        "author": 1,
        "cuisine": 1,
        "course": 2
      },
      {
        "id": 1,
        "photo": null,
        "photo_thumbnail": null,
        "ingredient_groups": [
          {
            "id": 1,
            "title": "Veges",
            "ingredients": [
              {
                "id": 1,
                "quantity": 1,
                "measurement": "dash",
                "title": "black pepper",
                "customQuantity": 1,
                "checked": false
              },
              {
                "id": 2,
                "quantity": 4,
                "measurement": "tablespoons",
                "title": "chili powder",
                "customQuantity": 4,
                "checked": false
              },
              {
                "id": 3,
                "quantity": 1,
                "measurement": "tablespoon",
                "title": "cumin",
                "customQuantity": 1,
                "checked": false
              },
              {
                "id": 4,
                "quantity": 1,
                "measurement": "can",
                "title": "dark kidney beans",
                "customQuantity": 1,
                "checked": false
              },
              {
                "id": 5,
                "quantity": 2,
                "measurement": "cans",
                "title": "diced tomatos",
                "customQuantity": 2,
                "checked": false
              },
              {
                "id": 6,
                "quantity": 1,
                "measurement": "whole",
                "title": "green bell pepper",
                "customQuantity": 1,
                "checked": false
              },
              {
                "id": 10,
                "quantity": 1,
                "measurement": "can",
                "title": "light kidney beans",
                "customQuantity": 1,
                "checked": false
              },
              {
                "id": 11,
                "quantity": 1,
                "measurement": "whole",
                "title": "serrano pepper ",
                "customQuantity": 1,
                "checked": false
              },
              {
                "id": 12,
                "quantity": 1,
                "measurement": "whole",
                "title": "white onion",
                "customQuantity": 1,
                "checked": false
              }
            ]
          },
          {
            "id": 2,
            "title": "Beef",
            "ingredients": [
              {
                "id": 7,
                "quantity": 1,
                "measurement": "pound",
                "title": "ground pork",
                "customQuantity": 1,
                "checked": false
              },
              {
                "id": 8,
                "quantity": 1,
                "measurement": "pound",
                "title": "ground sirloin",
                "customQuantity": 1,
                "checked": false
              },
              {
                "id": 9,
                "quantity": 1,
                "measurement": "dash",
                "title": "kosher salt",
                "customQuantity": 1,
                "checked": false
              }
            ]
          }
        ],
        "tags": [],
        "subrecipes": [],
        "pub_date": "2011-05-21",
        "update_date": "2011-05-21",
        "username": "ryan",
        "title": "Tasty Chili",
        "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
        "directions": "Brown the ground pork and ground sirlion in a medium pan. Add a teaspoon of sereano pepper while browning the meat. Season with kosher salt and pepper.\nChop the onion, bell pepper and one Serrano pepper and place them in a large pot.\nOpen up and drain both cans of kidney beans and add them to the large pot.\nOpen up both cans of stewed chili style tomatoes and add them to the pot.\nDrain the grease away from the browned meat and add the meat to the pot.\nPour in the tomato juice over the meat mixture.\nAdd kosher salt, black pepper, two table spoons of chili powder, and two teaspoons of ground cumin. Stir well.\nCook slowly over medium low heat for an hour. If it starts to bubble turn down the heat. Taste during the cooking process to check the seasoning add more to taste.",
        "source": "",
        "prep_time": 60,
        "cook_time": 60,
        "servings": 8,
        "rating": 3,
        "public": true,
        "author": 1,
        "cuisine": 1,
        "course": 2
      },
      {
        "id": 6,
        "photo": null,
        "photo_thumbnail": null,
        "ingredient_groups": [
          {
            "id": 6,
            "title": "",
            "ingredients": [
              {
                "id": 20,
                "quantity": 0,
                "measurement": null,
                "title": "sdfsdf"
              }
            ]
          }
        ],
        "tags": [],
        "subrecipes": [],
        "pub_date": "2011-05-21",
        "update_date": "2018-01-30",
        "username": "ryan",
        "title": "Tasty Chili 114",
        "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
        "directions": "sdfsdsdf",
        "source": "",
        "prep_time": 60,
        "cook_time": 60,
        "servings": 8,
        "rating": 3,
        "public": true,
        "author": 1,
        "cuisine": 1,
        "course": 2
      },
      {
        "id": 27,
        "photo": null,
        "photo_thumbnail": null,
        "ingredient_groups": [],
        "tags": [],
        "subrecipes": [],
        "pub_date": "2011-05-21",
        "update_date": "2011-05-21",
        "username": "ryan",
        "title": "Tasty Ch15ili 4",
        "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
        "directions": "",
        "source": "",
        "prep_time": 60,
        "cook_time": 60,
        "servings": 8,
        "rating": 3,
        "public": true,
        "author": 1,
        "cuisine": 1,
        "course": 2
      },
      {
        "id": 4,
        "photo": null,
        "photo_thumbnail": null,
        "ingredient_groups": [
          {
            "id": 4,
            "title": "",
            "ingredients": [
              {
                "id": 14,
                "quantity": 0,
                "measurement": null,
                "title": "dfghdfgh"
              }
            ]
          }
        ],
        "tags": [],
        "subrecipes": [],
        "pub_date": "2011-05-21",
        "update_date": "2011-05-21",
        "username": "ryan",
        "title": "Tasty Chili 4",
        "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
        "directions": "",
        "source": "",
        "prep_time": 60,
        "cook_time": 60,
        "servings": 8,
        "rating": 3,
        "public": true,
        "author": 1,
        "cuisine": 1,
        "course": 2
      },
      {
        "id": 18,
        "photo": null,
        "photo_thumbnail": null,
        "ingredient_groups": [],
        "tags": [],
        "subrecipes": [],
        "pub_date": "2011-05-21",
        "update_date": "2011-05-21",
        "username": "ryan",
        "title": "Tasty Chili 54",
        "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
        "directions": "",
        "source": "",
        "prep_time": 60,
        "cook_time": 60,
        "servings": 8,
        "rating": 3,
        "public": true,
        "author": 1,
        "cuisine": 1,
        "course": 2
      },
      {
        "id": 2,
        "photo": null,
        "photo_thumbnail": null,
        "ingredient_groups": [],
        "tags": [],
        "subrecipes": [],
        "pub_date": "2011-05-21",
        "update_date": "2011-05-21",
        "username": "ryan",
        "title": "Tasty Chili 2",
        "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
        "directions": "",
        "source": "",
        "prep_time": 60,
        "cook_time": 60,
        "servings": 8,
        "rating": 3,
        "public": true,
        "author": 1,
        "cuisine": 1,
        "course": 2
      },
      {
        "id": 7,
        "photo": null,
        "photo_thumbnail": null,
        "ingredient_groups": [],
        "tags": [],
        "subrecipes": [],
        "pub_date": "2011-05-21",
        "update_date": "2011-05-21",
        "username": "ryan",
        "title": "Tasty Chili 1114",
        "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
        "directions": "",
        "source": "",
        "prep_time": 60,
        "cook_time": 60,
        "servings": 8,
        "rating": 3,
        "public": true,
        "author": 1,
        "cuisine": 1,
        "course": 2
      },
      {
        "id": 34,
        "photo": "http://localhost:8000/site-media/upload/recipe_photos/Sheet-Pan-BBQ-Meatloaf-Dinner-V2_Iyritb3.jpg",
        "photo_thumbnail": "http://localhost:8000/site-media/CACHE/images/upload/recipe_photos/Sheet-Pan-BBQ-Meatloaf-Dinner-V2_Iyritb3/c3eaf4db4b2bc29e514a92607ddb4904.jpg",
        "ingredient_groups": [
          {
            "id": 14,
            "title": "",
            "ingredients": [
              {
                "id": 33,
                "quantity": 0,
                "measurement": null,
                "title": "dfg",
                "customQuantity": 0,
                "checked": false
              }
            ]
          }
        ],
        "tags": [],
        "subrecipes": [],
        "pub_date": "2018-01-30",
        "update_date": "2018-01-30",
        "username": "ryan",
        "title": "dgf",
        "info": "dfg",
        "directions": "dfg",
        "source": "",
        "prep_time": 5,
        "cook_time": 5,
        "servings": 5,
        "rating": 5,
        "public": true,
        "author": 1,
        "cuisine": 9,
        "course": 4
      },
      {
        "id": 12,
        "photo": null,
        "photo_thumbnail": null,
        "ingredient_groups": [],
        "tags": [],
        "subrecipes": [],
        "pub_date": "2011-05-21",
        "update_date": "2011-05-21",
        "username": "ryan",
        "title": "Tasty Chili 3334",
        "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
        "directions": "",
        "source": "",
        "prep_time": 60,
        "cook_time": 60,
        "servings": 8,
        "rating": 3,
        "public": true,
        "author": 1,
        "cuisine": 1,
        "course": 2
      },
      {
        "id": 33,
        "photo": "http://localhost:8000/site-media/upload/recipe_photos/g3O1DsG.jpg",
        "photo_thumbnail": "http://localhost:8000/site-media/CACHE/images/upload/recipe_photos/g3O1DsG/c08dba895cc303fa83d51c6d95463a28.jpg",
        "ingredient_groups": [
          {
            "id": 12,
            "title": "",
            "ingredients": [
              {
                "id": 31,
                "quantity": 0,
                "measurement": null,
                "title": "asd"
              }
            ]
          }
        ],
        "tags": [],
        "subrecipes": [],
        "pub_date": "2018-01-30",
        "update_date": "2018-01-30",
        "username": "ryan",
        "title": "zdfgsdfg",
        "info": "234234",
        "directions": "aerter",
        "source": "",
        "prep_time": 4,
        "cook_time": 4,
        "servings": 4,
        "rating": 4,
        "public": true,
        "author": 1,
        "cuisine": 5,
        "course": 1
      },
      {
        "id": 26,
        "photo": "http://localhost:8000/site-media/upload/recipe_photos/20161119_190343.jpg",
        "photo_thumbnail": "http://localhost:8000/site-media/CACHE/images/upload/recipe_photos/20161119_190343/8150a7802d8c7d4a9901aae42820d808.jpg",
        "ingredient_groups": [
          {
            "id": 8,
            "title": "",
            "ingredients": [
              {
                "id": 22,
                "quantity": 0,
                "measurement": null,
                "title": "sdefs"
              }
            ]
          }
        ],
        "tags": [],
        "subrecipes": [],
        "pub_date": "2011-05-21",
        "update_date": "2018-01-30",
        "username": "ryan",
        "title": "Tasty Ch14ili 4",
        "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
        "directions": "fsdfsf",
        "source": "",
        "prep_time": 60,
        "cook_time": 60,
        "servings": 8,
        "rating": 3,
        "public": true,
        "author": 1,
        "cuisine": 1,
        "course": 2
      },
      {
        "id": 16,
        "photo": null,
        "photo_thumbnail": null,
        "ingredient_groups": [],
        "tags": [],
        "subrecipes": [],
        "pub_date": "2011-05-21",
        "update_date": "2011-05-21",
        "username": "ryan",
        "title": "Tasty Chili 444",
        "info": "This chili is requested every winter by friends and family. I have been  making this chili every since I was a small child and learned from my grandma",
        "directions": "",
        "source": "",
        "prep_time": 60,
        "cook_time": 60,
        "servings": 8,
        "rating": 3,
        "public": true,
        "author": 1,
        "cuisine": 1,
        "course": 2
      }
    ],
    "status": ""
  },
  "recipeForm": {
    "form": [
      {
        "id": 0,
        "public": true,
        "errors": []
      }
    ],
    "status": {
      "message": "",
      "alert": ""
    },
    "recipeGroups": {
      "courses": [
        {
          "id": 3,
          "title": "Appetizer",
          "slug": "appetizer",
          "author": 1
        },
        {
          "id": 1,
          "title": "Breakfast",
          "slug": "breakfast",
          "author": 1
        },
        {
          "id": 4,
          "title": "Dessert",
          "slug": "dessert",
          "author": 1
        },
        {
          "id": 2,
          "title": "Entry",
          "slug": "entry",
          "author": 1
        }
      ],
      "cuisines": [
        {
          "id": 1,
          "title": "American",
          "slug": "american",
          "author": 1
        },
        {
          "id": 2,
          "title": "Chinese",
          "slug": "chinese",
          "author": 1
        },
        {
          "id": 3,
          "title": "French",
          "slug": "french",
          "author": 1
        },
        {
          "id": 4,
          "title": "Indian",
          "slug": "indian",
          "author": 1
        },
        {
          "id": 5,
          "title": "Italian",
          "slug": "italian",
          "author": 1
        },
        {
          "id": 6,
          "title": "Japanese",
          "slug": "japanese",
          "author": 1
        },
        {
          "id": 7,
          "title": "Korean",
          "slug": "korean",
          "author": 1
        },
        {
          "id": 8,
          "title": "Mexican",
          "slug": "mexican",
          "author": 1
        },
        {
          "id": 9,
          "title": "Thai",
          "slug": "thai",
          "author": 1
        }
      ],
      "tags": []
    }
  }
}