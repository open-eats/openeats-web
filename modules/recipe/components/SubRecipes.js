import React from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

import { Checkbox } from '../../common/components/FormComponents'

const SubRecipes = ({ data, check }) => {
  let subRecipes = data.map((subRecipe, i) => {
    return (
      <li className="ingredient" key={ i }>
        <Checkbox
          name={ subRecipe.child_recipe_id }
          checked={ subRecipe.checked ? subRecipe.checked : false }
          change={ check }
        />
        { (subRecipe.quantity !== 0)
            ? <span className="quantity">{ subRecipe.quantity } </span>
            : null
        }
        { (subRecipe.measurement)
            ? <span className="measurement">{ subRecipe.measurement } </span>
            : null
        }
        { (subRecipe.title)
            ? <Link to={ "/recipe/" + subRecipe.slug } className="title">{ subRecipe.title }</Link>
            : null
        }
      </li>
    );
  });

  return (
    <ul className="ingredients" >
      { subRecipes }
    </ul>
  );
};

SubRecipes.PropTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({
    child_recipe_id: PropTypes.number.isRequired,
    slug: PropTypes.string.isRequired,
    quantity: PropTypes.number.isRequired,
    measurement: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  }).isRequired).isRequired
};

export default SubRecipes;
