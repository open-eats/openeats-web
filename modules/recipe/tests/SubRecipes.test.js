import React from 'react';
import SubRecipes from '../components/SubRecipes';
import createComponentWithIntlAndRouter from '../../../config/jest/createComponentWithIntlAndRouter';

import data from './data';

test('Test Sub Recipes', () => {
  const check = jest.fn();
  const component = createComponentWithIntlAndRouter(
    <SubRecipes data={ data.subrecipes } check={ check }/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
