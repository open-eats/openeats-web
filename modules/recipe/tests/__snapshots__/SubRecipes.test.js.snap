// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Sub Recipes 1`] = `
<ul
  className="ingredients"
>
  <li
    className="ingredient"
  >
    <div
      className={undefined}
    >
      <label
        className="check-container"
      >
        <input
          checked={false}
          name={1}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <span
      className="quantity"
    >
      1
       
    </span>
    <span
      className="measurement"
    >
      dash
       
    </span>
    <a
      className="title"
      href="/recipe/undefined"
      onClick={[Function]}
    >
      black pepper
    </a>
  </li>
  <li
    className="ingredient"
  >
    <div
      className={undefined}
    >
      <label
        className="check-container"
      >
        <input
          checked={false}
          name={2}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
    <span
      className="quantity"
    >
      4
       
    </span>
    <span
      className="measurement"
    >
      tablespoons
       
    </span>
    <a
      className="title"
      href="/recipe/undefined"
      onClick={[Function]}
    >
      chili powder
    </a>
  </li>
</ul>
`;
